### ISSUE TITLE
Can't make it start on Plex Synology ARMv7 package

### ISSUE THREAD

**Issue by tproenca** [trakt/Plex-Trakt-Scrobbler#293]
Hi,
I have a NAS running Debian 14.04 and Plex Synology ARMv7 package. Plex tries to start the plugin but after a while it kills it. I'm attaching the logs. 

Does anybody have this problem on ARMv7 ?

[Plex Media Server.log]
```
Mar 12, 2016 19:22:15 [0xab1ff400] DEBUG - [com.plexapp.plugins.trakttv] Setting plug-in to always running (daemon mode).
Mar 12, 2016 19:22:15 [0xab1ff400] DEBUG - Starting file watcher for com.plexapp.plugins.trakttv
Mar 12, 2016 19:22:15 [0xab1ff400] DEBUG - Spawned plug-in com.plexapp.plugins.trakttv with PID 1199
Mar 12, 2016 19:22:15 [0xab1ff400] DEBUG - [com.plexapp.plugins.trakttv] Sending command: GET /:/prefixes
Mar 12, 2016 19:22:21 [0xaa9ff400] DEBUG - Request: [127.0.0.1:50500] GET /system/messaging/clear_events/com.plexapp.plugins.trakttv (8 live) GZIP
Mar 12, 2016 19:22:21 [0xaa9ff400] DEBUG - [com.plexapp.system] Sending command over HTTP (GET): /system/messaging/clear_events/com.plexapp.plugins.trakttv
Mar 12, 2016 19:22:21 [0xaa9ff400] DEBUG - HTTP requesting GET http://127.0.0.1:34406/system/messaging/clear_events/com.plexapp.plugins.trakttv
Mar 12, 2016 19:22:21 [0xb29ff400] DEBUG - Completed: [127.0.0.1:50500] GET /system/messaging/clear_events/com.plexapp.plugins.trakttv (11 live) GZIP 512ms 180 bytes 200
Mar 12, 2016 19:22:33 [0xab1ff400] ERROR - Timeout or error reading status line from plug-in pipe [com.plexapp.plugins.trakttv], we're killing it.
Mar 12, 2016 19:22:33 [0xab1ff400] DEBUG - Stopping [com.plexapp.plugins.trakttv].
Mar 12, 2016 19:22:33 [0xab1ff400] DEBUG - [com.plexapp.plugins.trakttv] Killing.
Mar 12, 2016 19:22:33 [0xab1ff400] ERROR - Error received reading configuration for com.plexapp.plugins.trakttv
Mar 12, 2016 19:22:33 [0xab1ff400] DEBUG - Stopping [com.plexapp.plugins.trakttv].
```

[com.plexapp.plugins.trakttv.log]
```
2016-03-12 19:22:21,293 (-495b4000) :  INFO (core:349) - Starting framework core - Version: 2.6.3, Build: 7be11e1 (Tue Feb 23 02:40:35 UTC 2016)
2016-03-12 19:22:21,295 (-495b4000) :  DEBUG (core:361) - Using the elevated policy
2016-03-12 19:22:21,298 (-495b4000) :  DEBUG (core:450) - Starting runtime component.
2016-03-12 19:22:21,310 (-495b4000) :  DEBUG (core:450) - Starting caching component.
2016-03-12 19:22:21,313 (-495b4000) :  DEBUG (core:450) - Starting data component.
2016-03-12 19:22:21,315 (-495b4000) :  DEBUG (core:450) - Starting networking component.
2016-03-12 19:22:21,341 (-495b4000) :  DEBUG (networking:282) - Loaded HTTP cookies
2016-03-12 19:22:21,364 (-495b4000) :  DEBUG (networking:450) - Setting the default network timeout to 20.0
2016-03-12 19:22:21,372 (-495b4000) :  DEBUG (core:450) - Starting localization component.
2016-03-12 19:22:21,375 (-495b4000) :  INFO (localization:408) - Setting the default locale to en-us
2016-03-12 19:22:21,380 (-495b4000) :  DEBUG (localization:426) - Loaded en strings
2016-03-12 19:22:21,384 (-495b4000) :  DEBUG (core:450) - Starting messaging component.
2016-03-12 19:22:21,389 (-495b4000) :  DEBUG (core:450) - Starting debugging component.
2016-03-12 19:22:21,393 (-495b4000) :  DEBUG (core:450) - Starting services component.
2016-03-12 19:22:21,397 (-495b4000) :  DEBUG (core:450) - Starting myplex component.
2016-03-12 19:22:21,399 (-4e800ba0) :  DEBUG (networking:166) - Requesting 'http://127.0.0.1:32400/system/messaging/clear_events/com.plexapp.plugins.trakttv'
2016-03-12 19:22:21,401 (-495b4000) :  DEBUG (core:450) - Starting notifications component.
2016-03-12 19:22:23,181 (-495b4000) :  DEBUG (accessor:68) - Creating a new model access point for provider com.plexapp.plugins.trakttv in namespace 'metadata'
2016-03-12 19:22:23,233 (-495b4000) :  DEBUG (networking:166) - Requesting 'http://127.0.0.1:32400/:/plugins/com.plexapp.system/resourceHashes'
2016-03-12 19:22:23,866 (-4e800ba0) :  DEBUG (services:265) - Plug-in is not daemonized - loading services from system
2016-03-12 19:22:23,866 (-495b4000) :  DEBUG (runtime:1117) - Created a thread named 'load_all_services'
2016-03-12 19:22:23,875 (-4e800ba0) :  DEBUG (networking:166) - Requesting 'http://127.0.0.1:32400/:/plugins/com.plexapp.system/messaging/function/X0J1bmRsZVNlcnZpY2U6QWxsU2VydmljZXM_/Y2VyZWFsMQoxCmxpc3QKMApyMAo_/Y2VyZWFsMQoxCmRpY3QKMApyMAo_'
2016-03-12 19:22:23,880 (-495b4000) :  DEBUG (runtime:1117) - Created a thread named 'get_server_info'
2016-03-12 19:22:23,884 (-495b4000) :  DEBUG (core:150) - Finished starting framework core
2016-03-12 19:22:23,887 (-495b4000) :  DEBUG (core:560) - Loading plug-in code
2016-03-12 19:22:24,153 (-495b4000) :  DEBUG (core:566) - Finished loading plug-in code
2016-03-12 19:22:24,154 (-4f800ba0) :  DEBUG (networking:166) - Requesting 'http://127.0.0.1:32400'
2016-03-12 19:22:24,286 (-495b4000) :  DEBUG (__init__:18) - Using locale: en_US.UTF-8
2016-03-12 19:22:25,136 (-495b4000) :  DEBUG (fs_migrator:19) - Running migration: <fs_migrator.Clean object at 0xa3c7b0>
2016-03-12 19:22:25,169 (-4f800ba0) :  DEBUG (core:538) - Machine identifier is 34fbf1ab9376f142cbc159ce284aa9630f65e9d4
2016-03-12 19:22:25,172 (-4f800ba0) :  DEBUG (core:539) - Server version is 0.9.15.6.1714-7be11e1
2016-03-12 19:22:26,195 (-4e800ba0) :  DEBUG (services:362) - Loaded services
2016-03-12 19:22:26,249 (-4f000ba0) :  DEBUG (services:438) - No shared code to load
2016-03-12 19:22:26,339 (-495b4000) :  DEBUG (fs_migrator:19) - Running migration: <fs_migrator.ForceLegacy object at 0xa3c810>
2016-03-12 19:22:26,355 (-495b4000) :  DEBUG (fs_migrator:19) - Running migration: <fs_migrator.SelectiveSync object at 0xa3c830>
2016-03-12 19:22:30,137 - plugin.core.logger.handlers.error_reporter.ErrorReporter (-495b4000) :  DEBUG (plugin.core.logger.handlers.error_reporter.ErrorReporter:194) - Configuring Raven for host: requests+http://d9114178666f4ca08d7d7babd8ba14a5:efc3c1947f25425eac24c8fe1a66ddde@sentry.skipthe.net:/1
2016-03-12 19:22:30,142 - plugin.core.logger.main          (-495b4000) :  DEBUG (plugin.core.logger.main:68) - Initialized logging (report: True, storage: False)
2016-03-12 19:22:32,962 - plugin.core.libraries.main       (-495b4000) :  INFO (plugin.core.libraries.main:28) - Using native libraries at 'Plug-in Support/Data/com.plexapp.plugins.trakttv/Libraries'
2016-03-12 19:22:32,966 - plugin.core.libraries.main       (-495b4000) :  DEBUG (plugin.core.libraries.main:144) - System: 'Linux', Architecture: 'armv7_hf'
2016-03-12 19:22:32,969 - plugin.core.libraries.helpers.path (-495b4000) :  DEBUG (plugin.core.libraries.helpers.path:30) - Inserted path: 'Plug-in Support/Data/com.plexapp.plugins.trakttv/Libraries/Linux/armv7_hf'
2016-03-12 19:22:32,972 - plugin.core.libraries.main       (-495b4000) :  DEBUG (plugin.core.libraries.main:178) - UCS: 'ucs4'
2016-03-12 19:22:32,975 - plugin.core.libraries.helpers.path (-495b4000) :  DEBUG (plugin.core.libraries.helpers.path:30) - Inserted path: 'Plug-in Support/Data/com.plexapp.plugins.trakttv/Libraries/Linux/armv7_hf/ucs4'
2016-03-12 19:22:32,978 - plugin.core.libraries.helpers.arm (-495b4000) :  DEBUG (plugin.core.libraries.helpers.arm:92) - Fetching processor information from "/proc/cpuinfo"...
2016-03-12 19:22:32,982 - plugin.core.libraries.helpers.system (-495b4000) :  INFO (plugin.core.libraries.helpers.system:171) - Executable at '' doesn't exist, using '/bin/ls' instead
2016-03-12 19:22:33,050 - plugin.core.libraries.main       (-495b4000) :  DEBUG (plugin.core.libraries.main:187) - CPU Type: '7-a'
2016-03-12 19:22:33,054 - plugin.core.libraries.main       (-495b4000) :  DEBUG (plugin.core.libraries.main:188) - Page Size: '4k'
2016-03-12 19:22:33,062 - plugin.core.libraries.main       (-495b4000) :  INFO (plugin.core.libraries.main:40) - [PATH] Plug-in Support/Data/com.plexapp.plugins.trakttv/Libraries/Linux/armv7_hf/ucs4
2016-03-12 19:22:33,065 - plugin.core.libraries.main       (-495b4000) :  INFO (plugin.core.libraries.main:40) - [PATH] Plug-in Support/Data/com.plexapp.plugins.trakttv/Libraries/Linux/armv7_hf
2016-03-12 19:22:33,071 - plugin.core.libraries.main       (-495b4000) :  INFO (plugin.core.libraries.main:40) - [PATH] Plug-ins/Trakttv.bundle/Contents/Libraries/Shared
2016-03-12 19:22:33,074 - plugin.core.libraries.main       (-495b4000) :  INFO (plugin.core.libraries.main:40) - [PATH] Plug-ins/Trakttv.bundle/Contents/Libraries/Linux/armv5tel
2016-03-12 19:22:33,077 - plugin.core.libraries.main       (-495b4000) :  INFO (plugin.core.libraries.main:45) - Testing native library support...
```

**Comment by fuzeman** [comment]
Which version of the plugin are you using?

Could you also post the result of this command:
```shell
cat /proc/cpuinfo
```

**Comment by tproenca** [comment]
# cat /proc/cpuinfo
Processor	: ARMv7 Processor rev 1 (v7l)
processor	: 0
BogoMIPS	: 3.27

processor	: 1
BogoMIPS	: 3.27

processor	: 2
BogoMIPS	: 3.27

processor	: 3
BogoMIPS	: 3.27

Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xc05
CPU revision	: 1

Hardware	: ODROIDC
Revision	: 000a
Serial		: 1b00000000000000

**Comment by fuzeman** [comment]
Thanks, will let you know when I've got a fix ready for testing.

**Comment by fuzeman** [comment]
I'm guessing you are using the Synology version of Plex Media Server on an ODROID device? *(at least, that's what it looks like from the cpu information you posted)*

We can't really identify what version of PMS you are using, so in order to "activate" the correct libraries in the plugin you will need to do the following:

1. Install this version of the plugin: https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/issue/293.zip

2. Create the file ```Plug-in Support/Data/com.plexapp.plugins.trakttv/advanced.ini``` with the following contents:

        [libraries]
        cpu_type = marvell-pj4

3. Restart Plex Media Server

----------------------------------------

Post the latest plugin log file again if it's still failing to startup correctly, have no idea if this will work correctly yet..

**Comment by tproenca** [comment]
Hi fuzeman,

I updated my server to 0.9.16.4 and download the zip file you suggested and created the ini file. Now is working like a charm. Thanks so much for your help with this. Much appreciated.

-- Tiago

**Comment by fuzeman** [comment]
Changes released in [v1.0.2.0](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.0.2.0)
